src/python_framework_v02/troute/network/reservoirs/hybrid/hybrid.pyx

```diff
@@ -0,0 +1,271 @@
+  )
+  void free_hybrid_reach(_Reach* reach)
+  void route(_Reach* reach, float routing_period, float inflow, float lateral_inflow, float* outflow,  float* water_elevation) nogil
+
+cdef void run(_Reach* reach, float inflow, float lateral_inflow, float routing_period, float* outflow,  float* water_elevation) nogil:
+    route(reach, inflow, lateral_inflow, routing_period, outflow, water_elevation)
+
+cdef class MC_Hybrid(Reach):
+  """
+    MC_Reservoir is a subclass of MC_Reach_Base_Class
+  """
+
+  def __init__(self, long id, int lake_number, long[::1] upstream_ids, args):
+    """
+      Construct the kernel based on passed parameters,
+      which only constructs the parent class
+
+      Params:
+        id: long
+          unique identity of the reach this reservoir represents
+        lake_number: int TODO (long?)
+          WRF_Hydro lake number of this reservoir
+        upstream_ids: array[long]
+          buffer/array of upstream identifiers which contribute flow to this reservoir
+        args: list
+          the hybrid parameters ordered as follows:
+            area = args[0]
+            max_depth = args[1]
+            orifice_area = args[2]
+            orifice_coefficient = args[3]
+            orifice_elevation  =  args[4]
+            weir_coefficient = args[5]
+            weir_elevation = args[6]
+            weir_length = args[7]
+            initial_fractional_depth  = args[8]
+            reservoir_type =  args[11]
+            reservoir_parameter_file = args[12]
+            start_date = args[13]
+            usgs_timeslice_path = args[14]
+            usace_timeslice_path = args[15]
+            observation_lookback_hours = args[16]
+            observation_update_time_interval_seconds = args[17]
+    """
+    super().__init__(id, upstream_ids, compute_type.RESERVOIR_HYBRID)
+    # Note Some issues with __calloc__:
+    # The python type isn't guaranteed to be properly constructed, so cannot depend on super class being constructured.
+    # Thus I don't think we can put these C init functions in __calloc__, at least not in all cases.
+    # init the backing struct, pass a dam_length of 10.0 for now
+
+    #Setting default dam_length to 10
+    dam_length = 10.0
+    area = args[0]
+    max_depth = args[1]
+    orifice_area = args[2]
+    orifice_coefficient = args[3]
+    orifice_elevation  =  args[4]
+    weir_coefficient = args[5]
+    weir_elevation = args[6]
+    weir_length = args[7]
+    initial_fractional_depth = args[8]
+    water_elevation = args[10]
+    reservoir_type = args[11]
+    reservoir_parameter_file = args[12]
+    start_date = args[13] 
+    usgs_timeslice_path = args[14] 
+    usace_timeslice_path = args[15]
+    observation_lookback_hours = args[16]
+    observation_update_time_interval_seconds = args[17]
+
+    #Check lengths of input strings to ensure that they do not exceed buffer size
+    if (len(reservoir_parameter_file) > 255):

```

Comment:
@hellkite500 Could you verify these boundary checks from our discussion from the other day?